Update Nix development and build guide (README.md) (#2873)
PostgREST/postgrest: nix/README.md

## Getting started with Nix

> You'll need to [get Nix](https://nixos.org/download.html). Please follow the recommended installations from the official nix download website unless you know what you are doing.

Reviewer: ```suggestion
You'll need to [get Nix](https://nixos.org/download.html). Follow the recommended installation for your operating system from the official download website.
```

I think the rest is not necessary, the more experienced Nix devs will ignore this anyways.

## Building PostgREST

To build PostgREST from your local checkout of the repository, run:

```bash

> $ nix-build --attr postgrestPackage

Reviewer: Cool. There are still other bash scripts that need this `$ ` before the command, it would be good to add them too. Also the ones starting with `postgrest-...` should have the `[nix-shell]$ ` before the command. This would make it clearer.
